Automattic/happychat-client
# Create npm package out of client's api

**Built on top of https://github.com/Automattic/happychat-client/pull/180**

## Summary
This PR prepares `happychat-client` to be exposed as a npm package. Paths are resolved using `babel-plugin-module-resolver` and configuration json files are inlined using `babel-plugin-inline-json`.

CSS url is moved to a configuration field and it will be server by webpack even when developing the npm package.

## Testing
1. Switch calypso to `try/happychat-client` branch (https://github.com/Automattic/wp-calypso/pull/23630)
2. Link this package to wp-calypso, by following these steps
- run `npm link` in this folder
- run `npm link happychat-client` in calypso folder
3. Run `npm run dev:standalone` in this repo
4. Grab your oauth token from a wpcom request made by the client (http://localhost:9000) it is under `Authorization` header after `Bearer`
5. Add the token to calypso https://github.com/Automattic/wp-calypso/pull/23630/files#diff-56de7193249f7b0107d6033c1a7737acR22
6. Run `npm dev:npm`
7. Access http://calypso.localhost:3000/me/chat
8. It should load the happychat-client iframe
